content/blog/2022/05/camel317-whatsnew/index.md

```diff
@@ -0,0 +1,114 @@
+to list used configurations for auto-configuration and property placeholders.
+
+The information displayed will show the source for each configuration; in other words
+you can exactly see if the configuration is from application.properties, OS environment variable,
+command line argument, or elsewhere.
+
+### Camel JBang (Camel CLI)
+
+In this release we have focused the most new development on the effort for Camel JBang.
+This is a new way of getting started and using Apache Camel.
+
+Users that are familiar with Camel K, and its "easy of use", then we are taking that
+same user experience to everyone; without the need for a Kubernetes platform.
+
+Having JBang installed, then you can easily run a Camel application from a terminal.
+
+To quickly create a new basic Camel integration use the `init` command:
+
+    camel init hello.java
+
+And then you can run the file:
+
+    camel run hello.java
+
+We made it easy to run existing examples directly loaded from GitHub:
+
+    camel run https://github.com/apache/camel-kamelets-examples/tree/main/jbang/hello-java
+
+And we also made it easy to download the example to local disk, so you can edit the source and
+see live updates, using the `init` command.
+
+    camel init https://github.com/apache/camel-kamelets-examples/tree/main/jbang/hello-java
+
+And then you can run in developer mode that has live reloading on source changes (on save):
+
+    camel run * --dev
+
+We continue our effort on Camel JBang for the upcoming 3.18 LTS release.
+
+There is a lot more that Camel JBang can do, so make sure to see the [Camel JBang documentation](/manual/camel-jbang.html).
+There you can also find information how to get JBang installed, and after that install the camel app in JBang.
+
+### Component Headers
+
+We completed the work of documenting every header every component supports.
+
+This means you can see in the website documentation what headers a component consumer or producer
+uses, and its purpose.
+
+This information is also available for tooling. For example the Camel IDEA plugin supports this
+and can do code assistance in your Camel routes.
+
+### Resume from Offset
+
+TODO: Otavio anything to mention (pausable/resumable)
+
+### Java DSL in Camel K
+
+The `camel-java-joor-dsl` used by Camel K, and Camel JBang, now supports compiling all Java sources
+in the same compilation unit, meaning they have the same classloader visibility and can use
+dependency injection among themselves. 
+
+### Camel YAML DSL
+
+The YAML parser now have precise source:line error reporting, meaning you can see exactly
+where there is a problem.
+
+### Load properties from valut/secrets cloud services
+
+We added support for Azure key vault service.
+See more details at the following [blog post](/blog/2022/03/secrets-properties-functions/)
+
+### Camel Kafka
+
+TODO: More Kafka ???
+
+We did other bug fixes, and we continue to make the camel-kafka component more robust. 1
```

Comment:
@davsclaus I'll try to complement this today afternoon. We have a few fixes and changes affecting this component.

Reply:
I'd like to mention that we support kafka transaction with https://issues.apache.org/jira/browse/CAMEL-15016

Reply:
@zhfeng 3.17 is being released today, do you have time to submit the kafka transaction text?

Reply:
@davsclaus yeah, I will submit it. sorry for the delay!

Reply:
@davsclaus I added it.